backend/core/src/Core.API/ResponseHandling/SignatureVerificationMiddleware.cs

```diff
@@ -0,0 +1,186 @@
+        {
+            try
+            {
+                // Retrieve headers from the request
+                string? signatureHeader = context.Request.Headers["x-signature"];
+                string? payloadHeader = context.Request.Headers["x-payload"];
+
+                // Retrieve method from the request
+                var method = context.Request.Method;
+
+                if (string.IsNullOrWhiteSpace(payloadHeader))
+                {
+                    _logger.LogError("Missing payload header");
+                    var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "x-payload"));
+                    await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                    return;
+                }
+
+                if (string.IsNullOrWhiteSpace(signatureHeader))
+                {
+                    _logger.LogError("Missing x-signature header");
+                    var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "x-signature"));
+                    await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                    return;
+                }
+
+                byte[] payloadBytes = Encoding.UTF8.GetBytes(payloadHeader);
+
+                JObject payloadJson = JObject.Parse(payloadHeader);
+
+                string? publicKey = null;
+                string? postPayload = null;
+
+                if (method == "POST" || method == "PUT")
+                {
+                    // Check if the "PostPayload" property is present
+                    if (payloadJson.TryGetValue(SignaturePayload.PostPayload, out var post))
+                    {
+                        postPayload = (string?)post;
+                    }
+                    else
+                    {
+                        _logger.LogError("Missing postPayload header");
+                        var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "postPayload"));
+                        await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                        return;
+                    }
+                }
+
+                // Check if the "publicKey" property is present
+                if (payloadJson.TryGetValue(SignaturePayload.PublicKey, out var pubKey))
+                {
+                    publicKey = (string?)pubKey;
+                }
+                else
+                {
+                    _logger.LogError("Missing publicKey header");
+                    var customErrors = new CustomErrors(new CustomError("Forbidden", "Missing Header", "publicKey"));
+                    await WriteCustomErrors(context.Response, customErrors, (int)HttpStatusCode.Forbidden);
+                    return;
+                }
+
+                // Get the current Unix UTC timestamp (rounded to 30 seconds)
```

Comment:
this still does not round to 30 seconds. Is the comment wrong or the implementation?

Reply:
Right, it missed the rounding part. thanks for checking it.

Reply:
Added it